src/task_types/burn_in_tests.rs

```diff
@@ -0,0 +1,334 @@
+            let mut params = self
+                .config_extraction_service
+                .task_def_to_resmoke_params(task_def, false)?;
+            params.resmoke_args = format!(
+                "{} {} {}",
+                params.resmoke_args, BURN_IN_REPEAT_CONFIG, &test
+            );
+            if params.generate_multiversion_combos {
+                for (old_version, version_combination) in self
+                    .multiversion_service
+                    .multiversion_iter(&params.suite_name)?
+                {
+                    let multiversion_name = self.multiversion_service.name_multiversion_suite(
+                        &params.suite_name,
+                        &old_version,
+                        &version_combination,
+                    );
+                    let multiversion_tags = Some(old_version.clone());
+
+                    let burn_in_suite_info = BurnInSuiteInfo {
+                        build_variant,
+                        total_tests: discovered_task.test_list.len(),
+                        task_name: &discovered_task.task_name,
+                        multiversion_name: Some(&multiversion_name),
+                        multiversion_tags,
+                    };
+
+                    sub_suites.push(self.create_task(&params, index, test, &burn_in_suite_info));
+                }
+            } else {
+                let burn_in_suite_info = BurnInSuiteInfo {
+                    build_variant,
+                    total_tests: discovered_task.test_list.len(),
+                    task_name: &discovered_task.task_name,
+                    multiversion_name: None,
+                    multiversion_tags: None,
+                };
+                sub_suites.push(self.create_task(&params, index, test, &burn_in_suite_info))
+            }
+        }
+
+        Ok(sub_suites)
+    }
+
+    /// Create an individual execution task for burn_in.
+    ///
+    /// # Arguments
+    ///
+    /// * `params` - Configuration for how suite should be configured.
+    /// * `index` - Index of sub-task in list of sub-tasks being created for the task.
+    /// * `test` - Name of test to generate sub-suite for.
+    /// * `suite_info` - Information about the suite being generated.
+    ///
+    /// # Returns
+    ///
+    /// Shrub task representing the given sub-task.
+    fn create_task(
+        &self,
+        params: &ResmokeGenParams,
+        index: usize,
+        test: &str,
+        suite_info: &BurnInSuiteInfo,
+    ) -> EvgTask {
+        let origin_suite = suite_info.build_origin_suite(&params.suite_name);
+
+        let sub_suite = SubSuite {
+            index: Some(index),
+            name: suite_info.build_display_name(),
+            test_list: vec![test.to_string()],
+            exclude_test_list: None,
+            origin_suite: origin_suite.clone(),
+            mv_exclude_tags: suite_info.multiversion_tags.clone(),
```

Comment:
Did we do this thing with adding tags in the old burn in task generator?
If this is just keeping the same functionality, then it's fine. I just don't remember, if resmoke is able to exclude the test from running, if we pass the test path as an argument.

Reply:
This could be a problem that needs more investigation. I don't think burn_in has been running multiversion tests properly, so we likely haven't hit this before. But we should investigate how resmoke will behave. Ideally, resmoke would just no-op in this case, which is what I think we would want.

Reply:
I've checked `--excludeWithAnyTags` resmoke's flag, the resmoke no-oped and didn't run the test even if I passed it as an argument. So if `mv_exclude_tags` end up being in `--excludeWithAnyTags` list, then it should work.